### PR TITLE
Add CFT (Cloud Foundation Toolkit) Builder

### PR DESCRIPTION
**Pull request by henryken**
This is to add a Cloud Builder for [CFT (Cloud Foundation Toolkit)](https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/cloud-foundation/community/cloud-foundation/docs/userguide.md).

### CODE REVIEW

**Comment by henryken** [comment]
This PR is to add cft builder as described in #137

**Comment by ocsig** [comment]
Great solution, I will test it tomorrow.
What we implemented for Jenkins and really want to make it work with cloud builder is dynamically generate build steps, so each build step is a deployment. This also gives potential for parallel execution and potential great UI support where the steps are shown separately and not just 1 giant console log.

The following command returns an execution order in a structured yaml format (you pass an entire directory of YAMLs where each file is a deployment with potential dependencies for other deployments):
`cft apply ${config_dir} --show-stages --format yaml> .stages.yaml`
Then based on the stages file you create n Cloudbuild Stages.

Here is the [Jenkins ](https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/master/community/cloud-foundation/pipeline/network/Jenkinsfile )implementation.

File: cft/examples/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+Note: This example assumes that you have built the `cft` build step and pushed it to
+`gcr.io/$PROJECT_ID/cft`.
+
+## Executing the CFT Builder
+
+To run this example, make sure you have assigned the [Deployment Manager
+Editor](https://cloud.google.com/iam/docs/understanding-roles#deployment_manager_roles)
+role to [your Cloud Build service
+account](https://cloud.google.com/cloud-build/docs/securing-builds/set-service-account-permissions),
```
Comment (nof20) [inline cft/examples/README.md]: Can we make this a bit easier for less experienced folks and list the exact commands?  I believe it is:

```bash
export PROJECT=$(gcloud info --format='value(config.project)')
export MEMBER=$(gcloud projects describe $PROJECT --format 'value(projectNumber)')@cloudbuild.gserviceaccount.com
gcloud projects add-iam-policy-binding $PROJECT --member=serviceAccount:$MEMBER --role='roles/deploymentmanager.editor'
```

File: cft/examples/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+and run:
+```
+    gcloud builds submit . --config=cloudbuild.yaml
+```
```
Comment (nof20) [inline cft/examples/README.md]: Can I suggest we add a clean-up step at the end too, e.g.

```
gcloud -q compute instances delete cft-instance-vm
```

**Comment by henryken** [comment]
Thanks for your review @nof20. I just added the changes based on your review comments.

File: cft/examples/cloudbuild.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Perform a cft command execution based on the `instance.yaml` configuration. This template
+# build creates a GCE image.
+#
+# See README.md for invocation instructions.
+steps:
+- name: 'gcr.io/$PROJECT_ID/cft'
+  args:
```
Comment (nof20) [inline cft/examples/cloudbuild.yaml]: Sorry - this doesn't work as currently formatted (I suspect it's `- --`).  Replace with:

```yaml
steps:
- name: 'gcr.io/$PROJECT_ID/cft'
  args: [ '--project=$PROJECT_ID',
          'apply',
          'instance.yaml' ]
```

**Pull request merged**